newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js

```diff
@@ -89,420 +136,746 @@ const getExtraInfoArray = (property: gdNamedPropertyDescriptor) => {
-                            value={
-                              property.getValue() === 'true' ? 'true' : 'false'
-                            }
-                            onChange={(e, i, value) => {
-                              property.setValue(value);
-                              this.forceUpdate();
-                              this.props.onPropertiesUpdated &&
-                                this.props.onPropertiesUpdated();
-                            }}
-                            fullWidth
-                          >
-                            <SelectOption
-                              value="true"
-                              label={t`True (checked)`}
-                            />
-                            <SelectOption
-                              value="false"
-                              label={t`False (not checked)`}
-                            />
-                          </SelectField>
-                        )}
-                        {property.getType() === 'Behavior' && (
-                          <BehaviorTypeSelector
-                            project={this.props.project}
-                            objectType={this.props.behaviorObjectType || ''}
-                            value={
-                              property.getExtraInfo().size() === 0
-                                ? ''
-                                : property.getExtraInfo().at(0)
-                            }
-                            onChange={(newValue: string) => {
-                              // Change the type of the required behavior.
-                              const extraInfo = property.getExtraInfo();
-                              if (extraInfo.size() === 0) {
-                                extraInfo.push_back(newValue);
-                              } else {
-                                extraInfo.set(0, newValue);
-                              }
-                              this.forceUpdate();
-                              this.props.onPropertiesUpdated &&
-                                this.props.onPropertiesUpdated();
-                            }}
-                            disabled={false}
-                          />
-                        )}
-                        {property.getType() === 'Color' && (
-                          <ColorField
-                            floatingLabelText={<Trans>Default value</Trans>}
-                            disableAlpha
-                            fullWidth
-                            color={property.getValue()}
-                            onChange={color => {
-                              property.setValue(color);
-                              this.forceUpdate();
-                              this.props.onPropertiesUpdated &&
-                                this.props.onPropertiesUpdated();
+  const pasteProperties = React.useCallback(
+    async effectInsertionIndex => {
```

Comment:
name to update

Reply:
Sorry, I though I did a pass on this. I must have been distracted.